components/tools/OmeroWeb/test/integration/test_thumbnails.py

```diff
@@ -112,3 +122,88 @@ def test_base64_thumb_set(self):
         json_data = json.loads(b64rsp)
         for i in images:
             assert json_data[str(i)] == expected_thumbs[i]
+
+
+class TestRoiThumbnails(IWebTest):
+
+    def shapes(self):
+        """Create a bunch of unsaved Shapes."""
+        rect = RectangleI()
+        rect.x = rdouble(10)
+        rect.y = rdouble(20)
+        rect.width = rdouble(30)
+        rect.height = rdouble(40)
+        rect.textValue = rstring("test-Rectangle")
+        rect.fillColor = rint(rgba_to_int(255, 255, 255, 255))
+        rect.strokeColor = rint(rgba_to_int(255, 255, 0, 255))
+
+        ellipse = EllipseI()
+        ellipse.x = rdouble(33)
+        ellipse.y = rdouble(44)
+        ellipse.radiusX = rdouble(55)
+        ellipse.radiusY = rdouble(66)
+        ellipse.textValue = rstring("test-Ellipse")
+
+        line = LineI()
+        line.x1 = rdouble(20)
+        line.x2 = rdouble(30)
+        line.y1 = rdouble(40)
+        line.y2 = rdouble(50)
+        line.textValue = rstring("test-Line")
+
+        point = PointI()
+        point.x = rdouble(50)
+        point.y = rdouble(50)
+        point.textValue = rstring("test-Point")
+
+        polygon = PolygonI()
+        polygon.fillColor = rint(rgba_to_int(255, 0, 255, 50))
+        polygon.strokeColor = rint(rgba_to_int(255, 255, 0))
+        polygon.strokeWidth = LengthI(10, UnitsLength.PIXEL)
+        points = "10,20, 50,150, 100,100, 150,75"
+        polygon.points = rstring(points)
+
+        polyline = PolylineI()
+        polyline.points = rstring(points)
+
+        return [rect, ellipse, line, point, polygon, polyline]
+
+    @pytest.mark.parametrize("theT", [1, 0])
+    @pytest.mark.parametrize("theZ", [0, 1])
+    def test_roi_thumbnail(self, theT, theZ):
+        update_service = self.sf.getUpdateService()
+        img = self.create_test_image(size_x=125, size_y=125, size_z=2,
+                                     size_t=2, session=self.sf)
+
+        for s in self.shapes():
+            if theT is not None:
+                s.theT = rint(theT)
+            if theZ is not None:
+                s.theZ = rint(theZ)
+            roi = RoiI()
+            roi.addShape(s)
+            roi.setImage(img)
+            roi = update_service.saveAndReturnObject(roi)
+            shape = roi.copyShapes()[0]
+
+            # Test ROI thumbnail...
+            request_url = reverse('webgateway_render_roi_thumbnail',
+                                  kwargs={'roiId': roi.id.val})
+            rsp = get(self.django_client, request_url)
+            thumb_bytes = BytesIO(rsp.content)
+            try:
+                thumb = Image.open(thumb_bytes)
+            finally:
+                thumb_bytes.close()
+            assert thumb.size == (250, 166)
```

Comment:
where is that size defined?

Reply:
width defined at https://github.com/ome/omero-web/blob/ec07fe59bf3af6ae002d634675d59a4b0b679b18/omeroweb/webgateway/views.py#L495

and height defined at https://github.com/ome/omero-web/blob/ec07fe59bf3af6ae002d634675d59a4b0b679b18/omeroweb/webgateway/views.py#L574

Reply:
Might be worth extracting such [magic numbers](https://en.wikipedia.org/wiki/Magic_number_(programming)) out to a constant somewhere (at some point in the future)